ledger-core-ripple/idl/callback.djinni

```diff
@@ -0,0 +1,17 @@
+@extern "../../ledger-core/idl/core.yaml"
+
+# Callback triggered by main completed task, returning optional result as list of template type T.
+ListCallback = interface[T] +j +o +s +n {
+    # Method triggered when main task complete.
+    # @params result optional of type list<T>, non null if main task failed
+    # @params error optional of type Error, non null if main task succeeded
+    onCallback(result: optional<list<T>>, error: optional<Error>);
+}
+
+# Callback triggered by main completed task, returning optional result of template type T.
+Callback = interface[T] +j +o +s +n {
+    # Method triggered when main task complete.
+    # @params result optional of type T, non null if main task failed
+    # @params error optional of type Error, non null if main task succeeded
+    onCallback(result: optional<T>, error: optional<Error>);
+}
```

Comment:
If I understood correctly we have to create callback types for each specific coins ?

Reply:
Yeah… it’s ultra boring and I hope we can get rid of `djinni` sooner or later because of that…